alacritty_terminal/src/term/mod.rs

```diff
@@ -921,6 +1042,107 @@ impl<T> Term<T> {
         self.dynamic_title = config.dynamic_title();
         self.auto_scroll = config.scrolling.auto_scroll;
         self.grid.update_history(config.scrolling.history() as usize, &self.cursor.template);
+        self.damage = DamageRect::whole_grid(&self.grid);
+        self.damage_list.clear();
+    }
+
+    /// Retrieve current damage and reset
+    ///
+    /// In order to reduce the performance overhead of tracking damage, damage
+    /// is not tracked by logging the cursor position of every cell update, but
+    /// instead work by looking at cursor progress.
+    ///
+    /// This is done by creating a DamageRect that encapsulates the intial
+    /// cursor position. When damage is read out, the current cursor position
+    /// is added to the rect. Explicit cursor moves and wrap-around also add a
+    /// rect that contains both the cursor before and after the move.
+    ///
+    /// This leaves the normal input path without any damage tracking overhead.
+    pub fn get_damage(&mut self) -> Vec<DamageRect> {
+        let cols = self.grid.num_cols();
+
+        let mut damage_list = std::mem::replace(&mut self.damage_list, Vec::new());
```

Comment:
I think `Vec::with_capacity(10)`(or for a previous `len`) as you're doing in init will make sense here too.